src/dispatch/common.rs

```diff
@@ -76,10 +76,9 @@ async fn missing_permissions<U, E>(
 /// Doesn't actually start the cooldown timer! This should be done by the caller later, after
 /// argument parsing.
 /// (A command that didn't even get past argument parsing shouldn't trigger cooldowns)
-#[allow(clippy::needless_lifetimes)] // false positive (clippy issue 7271)
 pub async fn check_permissions_and_cooldown<'a, U, E>(
     ctx: crate::Context<'a, U, E>,
-    cmd: &crate::Command<U, E>,
+    cmd: &'a crate::Command<U, E>,
```

Comment:
This is not equivalent to before. Like this, check_permissions_and_cooldown can't accept commands with a lifetime smaller than the Context lifetime anymore. https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=349063a911b6803bf8d2b5d1a46f8351